FWCore/Framework/src/globalTransitionAsync.h

```diff
@@ -181,46 +125,20 @@ namespace edm {
                                   [iWait, excpt](std::exception_ptr const*) mutable { iWait.doneWaiting(excpt); });
                               WaitingTaskHolder h(delayError);
                               for (auto& subProcess : iSubProcesses) {
-                                subProcessDoGlobalEndTransitionAsync(
-                                    h, subProcess, iPrincipal, iTS, iEventSetupImpls, cleaningUpAfterException);
+                                subProcessDoGlobalEndTransitionAsync(h, subProcess, info, cleaningUpAfterException);
                               }
                             } else {
                               for (auto& subProcess : iSubProcesses) {
-                                subProcessDoGlobalEndTransitionAsync(
-                                    iWait, subProcess, iPrincipal, iTS, iEventSetupImpls, cleaningUpAfterException);
+                                subProcessDoGlobalEndTransitionAsync(iWait, subProcess, info, cleaningUpAfterException);
                               }
                             }
                           });
 
     WaitingTaskHolder h(subs);
-    iSchedule.processOneGlobalAsync<Traits>(std::move(h), iPrincipal, iES, token, cleaningUpAfterException);
+    iSchedule.processOneGlobalAsync<Traits>(
+        std::move(h), transitionInfo.principal(), transitionInfo.eventSetupImpl(), token, cleaningUpAfterException);
```

Comment:
So the reason why the ProcessBlockTransitionInfo has a `eventSetupImpl()` method is because of the Schedule class? So we change it to also take a `TransitionInfo` as an argument?

Reply:
We could try that. I would have to look closer, but I think this goes all the way down the stack into the Worker methods like doWorkAsync. And then the other things that call doWorkAsync would be affected if we changed that interface. In the end this might be an improvement, but it would affect a lot of code.